{"version":1,"ops":[{"type":1,"author":{"id":"b83e8b4dfca7ebd9aba65d02ac0e5e24f95b7802"},"timestamp":1507238858,"metadata":{"github-id":"MDU6SXNzdWUyNjMyNzIwNTE=","github-url":"https://github.com/mindriot101/rust-fitsio/issues/41","origin":"github"},"title":"Test errors on OSX","message":"When running tests on OSX in parallel (i.e. by default, rather than with `cargo test -- --test-threads 1`), the tests either segfaults or complains about too many I/O drivers.\n\nWith lldb I can see that it's crashing in a stack of `libsystem_c.dylib` function calls:\n\n```\n(lldb) bt 7\n* thread #2, name = 'fitsfile::test::cannot_write_column_to_image_hdu', stop reason = EXC_BAD_ACCESS (code=1, address=0x68)\n  * frame #0: 0x00007fffc00576a0 libsystem_c.dylib`flockfile + 4\n    frame #1: 0x00007fffc005a83c libsystem_c.dylib`fwrite + 72\n    frame #2: 0x000000010030d231 libcfitsio.5.dylib`file_write + 112\n    frame #3: 0x000000010030bd16 libcfitsio.5.dylib`ffwrite + 50\n    frame #4: 0x0000000100300688 libcfitsio.5.dylib`ffbfwt + 346\n    frame #5: 0x00000001003010b4 libcfitsio.5.dylib`ffflsh + 58\n    frame #6: 0x00000001003032b1 libcfitsio.5.dylib`ffclos + 158\n```\n\nThis problem does not occur with single-threaded code, suggesting that it's a race condition or general threading bug in OSX?! The problem does not happen in my virtual machine, even when running with 4 cores and multiple threads. I do not have a proper physical linux machine to test on :(","files":null}]}